src/algebra/module/basic.lean

```diff
@@ -357,6 +358,47 @@ def add_comm_group.int_module.unique : unique (module ℤ M) :=
 { default := by apply_instance,
   uniq := λ P, module.ext' P _ $ λ n, int_smul_eq_zsmul P n }
 
+variables (f : ℕ → R) (g : ℕ → M) {m n : ℕ}
+open finset
+-- The partial sum of `g`, starting from zero
+local notation `G` n:80 := ∑ i in range n, g i
+
+
+/- **Summation by parts**, also known as **Abel's lemma** or an **Abel transformation**
+This is the version for modules, see `finset.sum_Ico_by_parts`` for a version in a commutative ring.
+-/
+theorem module.sum_Ico_by_parts (hmn : m < n) :
```

Comment:
Since this generalises the older `finset.sum_Ico_by_parts` I think we should just drop `finset.sum_Ico_by_parts`.

Furthermore `finset.sum_Ico_by_parts` redundantly requires a `comm_ring β` instead of just `ring β` because its statement multiplies by `G (i+1)` on the left instead of on the right. After fixing its statement, it has a one-line proof via this result: `by simp only [← smul_eq_mul, module.sum_Ico_by_parts f g hmn]`.

In fact could we invert the import changes and move this result to `algebra/big_operators/invervals.lean`?

(Similar remarks for `sum_range_by_parts` also of course.)

Reply:
I decided to try out my own suggestion but I'll have to wait some time to know if it passes CI so I pushed to a new branch in case it doesn't work out: branch#dirichlet-test

Reply:
It worked so I've pushed a commit doing this.